src/App/Settlements/sagas.ts

```diff
@@ -47,36 +59,402 @@ class FinalizeSettlementAssertionError extends Error {
+        const statePosition = stateOrder.indexOf(currentState);
+        const newStatePosition = stateOrder.indexOf(newState);
+        assert(
+          statePosition !== -1 && newStatePosition !== -1,
+          `Runtime error determining relative order of settlement participant account states ${newState}, ${currentState}`,
+        );
+        // If the settlement account state is already the target state, then we'll do nothing and
+        // exit here, returning null;
+        if (statePosition >= newStatePosition) {
+          return null;
+        }
+        if (adjustNdc) {
+          const ndcResult: ApiResponse = yield call(apis.participantLimits.update, {
+            participantName: adjustment.participant.name,
+            body: {
+              currency: adjustment.positionAccount.currency,
+              limit: {
+                ...adjustment.currentLimit,
+                value: adjustment.settlementBankBalance,
+              },
+            },
+          });
+          if (ndcResult.status !== 200) {
+            return {
+              type: FinalizeSettlementProcessAdjustmentsErrorKind.SET_NDC_FAILED,
+              value: {
+                adjustment,
+                error: ndcResult.data,
+              },
+            };
+          }
+        }
+
+        const description = `Business Operations Portal settlement ID ${settlement.id} finalization report processing`;
+        // We can't make a transfer of zero amount, so we have nothing to do. In this case, we can
+        // just skip the remaining steps.
+        if (adjustment.amount === 0 || !adjustLiquidityAccountBalance) {
+          // TODO: this is duplicated from below, is there a tidy way to rearrange the logic here?
+          // Set the settlement participant account to the new state
+          const spaResult = yield call(apis.settlementParticipantAccount.update, {
+            settlementId: settlement.id,
+            participantId: adjustment.settlementParticipant.id,
+            accountId: adjustment.settlementParticipantAccount.id,
+            body: {
+              state: newState,
+              reason: description,
+            },
+          });
+          if (spaResult.status !== 200) {
+            return {
+              type: FinalizeSettlementProcessAdjustmentsErrorKind.SETTLEMENT_PARTICIPANT_ACCOUNT_UPDATE_FAILED,
+              value: {
+                adjustment,
+                error: spaResult.data,
+              },
+            };
+          }
+          return 'OK';
+        }
+        // Make the call to process funds out, then poll the balance until it's reduced
+        const fundsInOutResult: ApiResponse = yield call(apis.participantAccount.create, {
+          participantName: adjustment.participant.name,
+          accountId: adjustment.settlementAccount.id,
+          body: {
+            externalReference: description,
+            action: adjustment.amount > 0 ? 'recordFundsIn' : 'recordFundsOutPrepareReserve',
```

Comment:
I can not find 'recordFundsOutCommit' after a 'recordFundsOutPrepareReserve' in the code.
I am not sure if it is the intended functionality.
I remember we used to do a put /participants/{name}/accounts/{id}/transfers/{transferId} for the recordFundsOutCommit after a post/participants/{name}/accounts/{id}

Reply:
Great catch, thank you :100: